src/main/java/org/ow2/proactive/catalog/repository/entity/KeyValueMetadataEntity.java

```diff
@@ -113,7 +109,7 @@ public boolean equals(Object o) {
     public int hashCode() {
         int result = super.hashCode();
         result = 31 * result + key.hashCode();
-        result = 31 * result + catalogObjectRevision.hashCode();
+        result = 31 * result + ((catalogObjectRevision == null) ? 0 : catalogObjectRevision.hashCode());
```

Comment:
When one puts the KeyValueMetadataEntity into a hashMap it fails because catalogObjectRevision can be null.